src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3363,18 +3240,10 @@ void c_typecheck_baset::typecheck_expr_binary_boolean(exprt &expr)
 void c_typecheck_baset::typecheck_side_effect_assignment(
   side_effect_exprt &expr)
 {
-  if(expr.operands().size()!=2)
-  {
-    error().source_location = expr.source_location();
-    error() << "operator '" << expr.get_statement() << "' expects two operands"
-            << eom;
-    throw 0;
-  }
-
   const irep_idt &statement=expr.get_statement();
 
-  exprt &op0=expr.op0();
-  exprt &op1=expr.op1();
+  exprt &op0 = to_binary_expr(expr).op0();
+  exprt &op1 = to_binary_expr(expr).op1();
```

Comment:
I think we should do `to_binary_expr` just once.

Reply:
done